File: client/src/app/shared/googlemap.service.ts
import { Injectable } from '@angular/core';
import {HttpClient, HttpHeaders, HttpParams} from "@angular/common/http";
import {environment} from "../../environments/environment.development";
import {lastValueFrom} from "rxjs";

@Injectable({
  providedIn: 'root'
})
export class GooglemapService {

  constructor(private http: HttpClient) { }

  GEOCODE_URL = "https://maps.googleapis.com/maps/api/geocode/json";

  PLACE_AUTOCOMPLETE_URL = "https://maps.googleapis.com/maps/api/place/autocomplete/json"

  //https://maps.googleapis.com/maps/api/geocode/json?key=AIzaSyArhQ2zZuRa7-QwSBIWs6L2SLQjEsx6QII&address=26 jalan berseh

  //https://maps.googleapis.com/maps/api/place/autocomplete/json?key=AIzaSyArhQ2zZuRa7-QwSBIWs6L2SLQjEsx6QII&input=26 jalan berseh&language=en&types=geocode


  getGeoCode(address: string) {

    // const params = new HttpParams()
    //   .append('key', environment.googleAPIKey)
    //   .append("address", address)
    //
    // return lastValueFrom(this.http.get(this.GEOCODE_URL, {params: params}))

    const params = new HttpParams()
      .append("key", environment.googleAPIKey)
      .append("input", address)
      .append("language", "en")
      .append("types", "geocode")
      .append("components", "country:sg")

    // const headers = new HttpHeaders()
    //   .append("Access-Control-Allow-Origin", "*")

    return lastValueFrom(this.http.get(this.PLACE_AUTOCOMPLETE_URL, {params: params}))
  }

}

File: server/src/main/java/com/project/meepletable/services/GoogleMapService.java
package com.project.meepletable.services;

import org.springframework.http.RequestEntity;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;
import org.springframework.web.util.UriComponentsBuilder;

@Service
public class GoogleMapService {

    public void getCoordinates(String address) {

// https://maps.googleapis.com/maps/api/place/autocomplete/json
// ?key=AIzaSyArhQ2zZuRa7-QwSBIWs6L2SLQjEsx6QII&input=26 jalan berseh&language=en&types=geocode

        final String GMAP_API = "https://maps.googleapis.com/maps/api/place/autocomplete/json";

        final String G_API_KEY = "AIzaSyArhQ2zZuRa7-QwSBIWs6L2SLQjEsx6QII";

        String url = UriComponentsBuilder.fromUriString(GMAP_API)
                .queryParam("key", G_API_KEY)
                .queryParam("input", address)
                .queryParam("language", "en")
                .queryParam("types", "geocode")
                .queryParam("components", "country:sg")
                .toUriString();

        RequestEntity<Void> request = RequestEntity.get(url).build();

        RestTemplate template = new RestTemplate();
        ResponseEntity<String> response = template.exchange(request, String.class);
        String payload = response.getBody();
        System.out.println(payload);



    }

}
